File: csharp/gabaritos/gabarito23.cs
using System;
using System.Collections.Generic;

class Pilha
{
    private List<int> items = new List<int>();

    public void Push(int item)
    {
        items.Add(item);
    }

    public int? Pop()
    {
        if (!IsEmpty())
        {
            int item = items[items.Count - 1];
            items.RemoveAt(items.Count - 1);
            return item;
        }
        return null;
    }

    public bool IsEmpty()
    {
        return items.Count == 0;
    }
}

class Program
{
    static void Main()
    {
        Pilha pilha = new Pilha();
        pilha.Push(1);
        pilha.Push(2);
        Console.WriteLine($"Pop: {pilha.Pop()}");
        Console.WriteLine($"Vazia: {pilha.IsEmpty()}");
    }
}

File: cpp/gabaritos/gabarito23.cpp
#include <iostream>
#include <vector>

class Pilha {
private:
    std::vector<int> items;
public:
    void push(int item) {
        items.push_back(item);
    }
    int pop() {
        if (!is_empty()) {
            int top = items.back();
            items.pop_back();
            return top;
        }
        return -1; // erro
    }
    bool is_empty() {
        return items.empty();
    }
};

int main() {
    Pilha pilha;
    pilha.push(1);
    pilha.push(2);
    std::cout << "Pop: " << pilha.pop() << std::endl;
    std::cout << "Vazia: " << (pilha.is_empty() ? "Sim" : "Não") << std::endl;
    return 0;
}